### DISCUSSION TITLE
Differences between versions.

### DISCUSSION THREAD

**Discussion by gonulbuya** [pixijs/pixijs#9443] [Ideas]
Hello. Pixijs is a very powerful and very good library.
But what he observes are differences between versions. The next version should support previous versions. Users should switch gradually. For example, code written in v5 does not work with v7. V7 and v5 codes should be able to be run. Someone with hundreds of files will have a hard time migrating from v5 to v7.

If you pay attention to this situation in the next versions, I think it will be very good. I wish you ease.

**Comment by miltoncandelero** [comment]
I disagree.

Besides, look at how Semantic Versioning works 

> Given a version number MAJOR.MINOR.PATCH, increment the

> MAJOR version when you make incompatible API changes

> MINOR version when you add functionality in a backward compatible manner

> PATCH version when you make backward compatible bug fixes

> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

It is to be expected a breaking change in a major release

**Reply by bigtimebuddy** [reply]
@gonulbuya I understand the frustration and expectation. Breaking changes are frustrating to deal with. My suggestion for users upgrading to v7 from v5, is to upgrade to the last v6 release _first_. Resolve any issues and deprecation warnings you see in the console using v6. This should make upgrading to v7 easier. Good luck.

**Reply by gonulbuya** [reply]
Thank you for your suggestion.
